packages/types/config/render.d.ts

```diff
@@ -7,20 +7,65 @@
  *                https://github.com/jshttp/etag#readme
  */
 
+import { IncomingMessage, ServerResponse } from 'http'
 import { CompressionOptions } from 'compression'
 import { Options as EtagOptions } from 'etag'
 import { ServeStaticOptions } from 'serve-static'
 import { BundleRendererOptions } from 'vue-server-renderer'
 import { NuxtConfigurationServerMiddleware } from './server-middleware'
 
+type NuxtEtagOptions = EtagOptions & {
+  hash?: (html: string) => string
+}
+
+type ServePlaceholderHandler = 'default' | 'css' | 'html' | 'js' | 'json' | 'map' | 'plain' | 'image'
+interface ServePlaceholderOptions {
+  handlers?: Record<string, ServePlaceholderHandler | null | false>
+  mimes?: Record<ServePlaceholderHandler, string | false | undefined>
+  noCache?: boolean
+  placeholders?: Record<ServePlaceholderHandler, string | Buffer | false>
+  skipUnknown?: boolean
+  statusCode?: false | number
+}
+
+type CspPolicyName = 'child-src' | 'connect-src' | 'default-src' | 'font-src' | 'frame-src' | 'img-src' | 'manifest-src' | 'media-src' | 'object-src' | 'prefetch-src' | 'script-src' | 'script-src-elem' | 'script-src-attr' | 'style-src' | 'style-src-elem' | 'style-src-attr' | 'worker-src' | 'base-uri' | 'plugin-types' | 'sandbox' | 'form-action' | 'frame-ancestors' | 'navigate-to' | 'report-uri' | 'report-to' | 'block-all-mixed-content' | 'referrer' | 'require-sri-for' | 'trusted-types' | 'upgrade-insecure-requests'
+interface CspOptions {
+  addMeta?: boolean
+  allowedSources?: {}
```

Comment:
Why `{}` ?

Reply:
Sorry, that was incomplete. Have updated to `string[]` which I think is correct though note what appears to be an incorrect test [here](https://github.com/nuxt/nuxt.js/blob/257ae2279b142962de928902fedfda29cb94bb7d/packages/config/test/options.test.js#L101).

Reply:
Hm alright, from what did you suppose it is `string[]` then ?

Reply:
[here](https://github.com/nuxt/nuxt.js/blob/4e4aa4d5d488a25bd2b78258c11b114808c9ea3d/packages/server/src/middleware/nuxt.js#L130), [here](https://github.com/nuxt/nuxt.js/blob/257ae2279b142962de928902fedfda29cb94bb7d/test/unit/basic.ssr.csp.test.js#L84) and [here](https://github.com/nuxt/nuxt.js/blob/4e4aa4d5d488a25bd2b78258c11b114808c9ea3d/packages/server/test/middleware/nuxt.test.js#L243)

Reply:
Cool thanks !